src/macaron/dependency_analyzer/java_repo_finder.py

```diff
@@ -182,30 +182,33 @@ def find_repo(group: str, artifact: str, version: str, tags: list[str]) -> Itera
     Parameters
     ----------
     group : str
-        The group identifier of an artifact
+        The group identifier of an artifact.
     artifact : str
-        The artifact name of an artifact
+        The artifact name of an artifact.
     version : str
-        The version number of an artifact
-    tags : list[str]
-        The list of XML tags to look for, each in the format: tag1[.tag2 ... .tagN]
+        The version number of an artifact.
+    tags : Iterator[str]
+        The list of XML tags to look for, each in the format: tag1[.tag2 ... .tagN].
 
     Returns
     -------
-    Iterator[str] :
+    list[str] :
```

Comment:
Why would this be changed to `list` when we are still returning an Iterator (e.g `return (iter[])`) ?

Reply:
I changed to a list as part of the earlier discussion, then reverted the change. Seems I missed a bit.